### ISSUE TITLE
Error running test case validity_period_test.go

### ISSUE THREAD

**Issue by srderson** [hyperledger-archives/fabric#524]
I hit the following error when running the entire test suite using the command

```
cd $GOPATH/src/github.com/openblockchain/obc-peer
go test -timeout=20m $(go list github.com/openblockchain/obc-peer/... | grep -v /vendor/)
```

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x6c1c06]

goroutine 22 [running]:
log.(*Logger).Output(0x0, 0x2, 0xc8201e6000, 0x3a, 0x0, 0x0)
    /opt/go/src/log/log.go:147 +0x4e6
log.(*Logger).Println(0x0, 0xc8201d3da0, 0x1, 0x1)
    /opt/go/src/log/log.go:182 +0x6a
github.com/openblockchain/obc-peer/obc-ca/obcca.startOpenchain(0x0, 0x0)
    /opt/gopath/src/github.com/openblockchain/obc-peer/obc-ca/obcca/validity_period_test.go:290 +0x3e2
github.com/openblockchain/obc-peer/obc-ca/obcca.startServices(0xc820190630)
    /opt/gopath/src/github.com/openblockchain/obc-peer/obc-ca/obcca/validity_period_test.go:138 +0x34
created by github.com/openblockchain/obc-peer/obc-ca/obcca.TestValidityPeriod
    /opt/gopath/src/github.com/openblockchain/obc-peer/obc-ca/obcca/validity_period_test.go:82 +0x5f

goroutine 1 [chan receive]:
testing.RunTests(0xe78e70, 0x130c950, 0x1, 0x1, 0x1)
    /usr/local/go/src/testing/testing.go:562 +0x8ad
testing.(*M).Run(0xc8200e5ee8, 0x4de529)
    /usr/local/go/src/testing/testing.go:494 +0x70
github.com/openblockchain/obc-peer/obc-ca/obcca.TestMain(0xc8200e5ee8)
    /opt/gopath/src/github.com/openblockchain/obc-peer/obc-ca/obcca/validity_period_test.go:63 +0x26
main.main()
    github.com/openblockchain/obc-peer/obc-ca/obcca/_test/_testmain.go:52 +0x113

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
    /opt/go/src/runtime/asm_amd64.s:1696 +0x1

goroutine 21 [sleep]:
time.Sleep(0x29e8d60800)
    /opt/go/src/runtime/time.go:59 +0xf9
github.com/openblockchain/obc-peer/obc-ca/obcca.TestValidityPeriod(0xc820190630)
    /opt/gopath/src/github.com/openblockchain/obc-peer/obc-ca/obcca/validity_period_test.go:85 +0x72
testing.tRunner(0xc820190630, 0x130c950)
    /usr/local/go/src/testing/testing.go:456 +0x98
created by testing.RunTests
    /usr/local/go/src/testing/testing.go:561 +0x86d

goroutine 23 [runnable]:
github.com/openblockchain/obc-peer/obc-ca/obcca.startTCA()
    /opt/gopath/src/github.com/openblockchain/obc-peer/obc-ca/obcca/validity_period_test.go:150
created by github.com/openblockchain/obc-peer/obc-ca/obcca.startServices
    /opt/gopath/src/github.com/openblockchain/obc-peer/obc-ca/obcca/validity_period_test.go:137 +0x2f
FAIL    github.com/openblockchain/obc-peer/obc-ca/obcca 0.039s
```

**Comment by jeffgarratt** [comment]
@srderson @diegomasini Is this still an issue?  Seems to pass for me.

**Comment by diegomasini** [comment]
It was fixed in #526. The test is skipped for now since as @srderson explained to me the tests require a peer running in a separated window. That conflicts with the peer launched as part of the validity_period test. The test is working if it is running alone (removing the skip sentence) without launching the additional peer in a separated window.

Adding @raullaprida to the conversation since he was working on this too.
